src/misc/messages.cpp

```diff
@@ -58,10 +58,11 @@ class Message {
 	                               std::map<uint16_t, std::string> &output_msg_by_codepage)
 	{
 		assert(msg.length());
-		const uint16_t cp = UTF8_GetCodePage();
+		const uint16_t cp = get_utf8_code_page();
 		if (output_msg_by_codepage[cp].empty()) {
-			if (!UTF8_RenderForDos(msg, output_msg_by_codepage[cp], cp))
+			if (!utf8_to_dos(msg, output_msg_by_codepage[cp], cp)) {
 				LOG_WARNING("LANG: Problem rendering string");
```

Comment:
maybe include the problematic strings when logging?

`LANG: Problem rendering string from '%s' to '%s'`

Reply:
Changed, but I'm tracing only the source string. Destination string might contain something the host OS (if it is UTF-8 aware) does not know how to handle.